### DISCUSSION TITLE
Python: need guidance applying OUTPUT to void **const

### DISCUSSION THREAD

**Discussion by mawenhua** [swig/swig#2133] [Q&A]
In interface file, I want to make the 4th parameter as OUTPUT
`dds_reg_pub(const uint8_t , const char *const , const char *const , void **const );`
but it seems I cannot directly apply it to OUTPUT, and when I try `typedef void * VOIDP;`
I got 
> TypeError: Wrong number or type of arguments for overloaded function 'dds_reg_pub'.
  Possible C/C++ prototypes are:
    dds_reg_pub(uint8_t const,char const *const,char const *const,VOIDP *const)
    dds_reg_pub(uint8_t const,char const *const,char const *const,void **const)
I tried kinds of combination of (%ignore, %rename, %apply), but either it still request for 4 args, or it cause overload, or the definition got problem. Can some experts help? Thanks in adv.

**Comment by ojwb** [comment]
Enabling github discussions was a bit of an experiment, but it doesn't seem to be working as a way for users to get help because very few people seem to read the discussions and so you're lucky if someone who can answer even sees your question.  If you still need help, the swig-user mailing list is probably a much better place to ask.

However I think you'll also probably need to explain what the function you're trying to wrap does, and in particular what it expects to be passed for the last parameter when called from C or C++ code.  Maybe it actually gives out a pointer of a particular type cast to `void*`, but what that type is depends on the values of some of the other parameters.  Or maybe this is giving you an opaque handle whose only use is to pass back to C/C++ via another function in the API being wrapped.  Or it could be something else again.
